### PR TITLE
SNOW-89150 Adding Validate Default Parameters

### PR DESCRIPTION
**Pull request by Mehuljain12**
SNOW-89150

### CODE REVIEW

**Comment by snowstakeda** [comment]
Please add JIRA number to the title, e.g., SNOW-NNNNN

File: lib/constants/error_messages.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ exports[404022] = 'Invalid region. The specified value must be a string.';
 exports[404023] = 'Invalid clientSessionKeepAlive. The specified value must be a boolean.';
 exports[404024] = 'Invalid clientSessionKeepAliveHeartbeatFrequency. The specified value must be a number.';
 exports[404025] = 'Invalid jsTreatIntegerAsBigInt. The specified value must be a boolean';
+exports[404026] = 'Invalid validateDefaultParameters. The specified value must be boolean';
```
Comment (snowstakeda) [inline lib/constants/error_messages.js]: As we discussed in the previous meeting, you don't need to add a error code for NodeJS but let the GS error arise.

File: package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "ocsp": "^1.2.0",
     "request": "^2.88.0",
     "simple-lru-cache": "^0.0.2",
+    "snowflake-sdk": "^1.5.0",
```
Comment (snowstakeda) [inline package.json]: Please remove this. 
This package.json is to install the dependencies and not the NodeJS connector itself.

File: test/integration/connectionOptions.js [hunk 3]
```diff
@@ -56,9 +56,10 @@ var valid =
     role: snowflakeTestRole
   };
 
+
 var snowflakeAccount = snowflakeTestAdminUser !== undefined ?
   {
-    accessUrl: accessUrl,
+    accessUrl: 'accessUrl',
```
Comment (snowstakeda) [inline test/integration/connectionOptions.js]: I don't believe you need accessUrl?

File: test/integration/connectionOptions.js [hunk 4]
```diff
@@ -67,7 +68,7 @@ var snowflakeAccount = snowflakeTestAdminUser !== undefined ?
 var wrongUserName =
   {
 
-    accessUrl: accessUrl,
+    accessUrl: 'accessUrl',
```
Comment (snowstakeda) [inline test/integration/connectionOptions.js]: Please remove this change.
This test is intended to detect the wrong user name error.

File: test/integration/testConnection.js [hunk 5]
```diff
@@ -98,8 +99,10 @@ describe('Connection test', function ()
       done();
     });
   });
+
   it('Multiple Client', function (done)
   {
+    this.timeout(15000);
```
Comment (snowstakeda) [inline test/integration/testConnection.js]: can you elaborate why you add this?

File: test/integration/testConnection.js [hunk 6]
```diff
@@ -110,25 +113,27 @@ describe('Connection test', function ()
...
+
       {
         testUtil.checkError(err);
         conn.execute({
           sqlText: 'select 1',
           complete: function (err)
           {
             testUtil.checkError(err);
-            testUtil.destroyConnection(conn, function ()
+            testUtil.destroyConnection(conn,function ()
```
Comment (snowstakeda) [inline test/integration/testConnection.js]: please keep a space

File: test/integration/testConnection.js [hunk 7]
```diff
@@ -110,25 +113,27 @@ describe('Connection test', function ()
...
         });
       });
     }
+
     setTimeout(function ()
     {
       assert.strictEqual(completedConnection, totalConnections);
       done();
-    }, 60000);
+    }, 9000);
```
Comment (snowstakeda) [inline test/integration/testConnection.js]: can you elaborate why you updated?

File: test/unit/snowflake_test.js [hunk 8]
```diff
@@ -1757,6 +1758,32 @@ describe('snowflake.createConnection() JS_TREAT_INTEGER_AS_BIGINT', function ()
   });
 });
 
+
+describe('snowflake.createConnection() VALIDATE_DEFAULT_PARAMETERS', function ()
```
Comment (snowstakeda) [inline test/unit/snowflake_test.js]: can you add test cases that raises the validation error and catch it?
You just need to pass a wrong database, schema or warehouse names and check the error code.

File: lib/constants/error_messages.js (hunk 1 above)
Comment (Mehuljain12) [reply lib/constants/error_messages.js]: Sure, will get that removed.

File: test/integration/testConnection.js (hunk 5 above)
Comment (Mehuljain12) [reply test/integration/testConnection.js]: I saw the multiple client test failing with 6000, so I increased the timeout to 15000 and that worked.  I am not sure, if it is to do with just my testing.

File: test/integration/connectionOptions.js (hunk 3 above)
Comment (Mehuljain12) [reply test/integration/connectionOptions.js]: sure. Thanks Shige

**Comment by notkriswagner** [comment]
Can you update the title of the code review to be a short description of the change you are trying to introduce?

**Comment by snowstakeda** [comment]
@Mehuljain12 please look into the test failures.

**Comment by sfc-gh-kwagner** [comment]
@Mehuljain12 can we close this PR?

**Comment by sfc-gh-kwagner** [comment]
@sfc-gh-stakeda can we close this PR?

**Comment by sfc-gh-stakeda** [comment]
yes, let's close this.

**Pull request closed without merging**